ROMFS/px4fmu_common/init.d/rc.mavlink

```diff
@@ -0,0 +1,280 @@
+#!nsh
+#
+# MAVLINK stream startup script.
+#
+# NOTE: Script variables are declared/initialized/unset in the rcS script.
+#
+# NOTE: Normal mode uses baud rate of 57600 (default) and data rate of 1000 bytes/s.
+#
+# NOTE: To utilize this startup script, set mavlink device stream settings in an
+#       airframe configuration file as shown here:
+#
+#       set SERIAL_DEVICE_2          "/dev/ttyS2"
+#       set SERIAL_DEVICE_2_ARGS     "-m onboard -r 10000 -x -f"
+#       set SERIAL_DEVICE_2_BAUD     "57600"
+#
+#------------------------------------------------------------------------------
+#
+# UART mapping on FMUv2/3/4:
+#
+# UART1         /dev/ttyS0      IO debug (except v4, there ttyS0 is the wifi)
+# USART2        /dev/ttyS1      TELEM1 (flow control)
+# USART3        /dev/ttyS2      TELEM2 (flow control)
+# UART4
+# UART7                         CONSOLE
+# UART8                         SERIAL4
+#
+#------------------------------------------------------------------------------
+#
+# UART mapping on FMUv5:
+#
+# UART1         /dev/ttyS0      GPS
+# USART2        /dev/ttyS1      TELEM1 (flow control)
+# USART3        /dev/ttyS2      TELEM2 (flow control)
+# UART4         /dev/ttyS3      TELEM4
+# USART6        /dev/ttyS4      TELEM3 (flow control)
+# UART7         /dev/ttyS5
+# UART8         /dev/ttyS6      CONSOLE
+#
+#------------------------------------------------------------------------------
+#
+# UART mapping on OMNIBUSF4SD:
+#
+# USART1        /dev/ttyS0      SerialRX
+# USART4        /dev/ttyS1      TELEM1
+# USART6        /dev/ttyS2      GPS
+#
+
+#
+# Setup for board specific configurations.
+#
+if ver hwcmp AEROFC_V1
+then
+	set SERIAL_DEVICE_1      "/dev/ttyS3"
+	set SERIAL_DEVICE_1_ARGS " -r 1200"
+
+	set SERIAL_DEVICE_3      none
+
+	# Only start mavlink if the Benewake TFMini or LeddarOne isn't being used.
+	if param greater SENS_EN_LEDDAR1 0
+	then
+		set SERIAL_DEVICE_1 none
+	fi
+
+	if param greater SENS_EN_TFMINI 0
+	then
+		set SERIAL_DEVICE_1 none
+	fi
+
+	if protocol_splitter start ${SERIAL_DEVICE_2}
+	then
+		mavlink start -d /dev/mavlink -b 921600 -m onboard -r 5000 -x
+		micrortps_client start -d /dev/rtps -b 921600 -l -1 -s 2000
+	fi
+fi
+
+if ver hwcmp CRAZYFLIE
+then
+	# Avoid using either of the two available serial devices.
+	set SERIAL_DEVICE_1 none
+fi
+
+if ver hwcmp NXPHLITE_V3
+then
+	set SERIAL_DEVICE_2 "/dev/ttyS4"
+fi
+
+if ver hwcmp MINDPX_V2 PX4FMU_V4PRO
+then
+	# This is TELEM4 on Pixhawk 3 Pro.
+	frsky_telemetry start -d /dev/ttyS6
+fi
+
+if ver hwcmp PX4FMU_V4
+then
+	# This is TELEM4 on Pixhawk 3 Pro.
+	frsky_telemetry start -d /dev/ttyS6
+
+	# Setup MAVLink on Wifi (ESP8266 port).
+	set SERIAL_DEVICE_0_ARGS "-r 20000"
+	set SERIAL_DEVICE_0_BAUD "921600"
+
+	# Use ttyS1 for MAVLink on FMUv4 in addition to ttyS0 (debug).
+	set SERIAL_DEVICE_1_ARGS "-r 1200"
+
+	# Start MAVLink on Wifi (ESP8266 port).
```

Comment:
There is something wrong here: we have this twice now, also for `SERIAL_DEVICE_0_BAUD` right above (which specifies the baudrate and args correctly).

Reply:
@bkueng Agreed.  I have dug into this and discovered additional cases that I am untangling now. Mavlink startups were the genesis of this PR, so it is fitting that they revisit me at this point in the work.  More work to follow.